src/components/util/AnotherBadComponent.tsx

```diff
@@ -0,0 +1,62 @@
+// AnotherBadComponent.tsx
+
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+const Container = styled.div`
+  padding: 20px;
+
+`;
+
+const Button = styled.button`
+  padding: 5px 10px;
+
+  color: white;
+`;
+
+const Label = styled.label`
+  font-size: 14px;
+  margin-right: 10px;
+`;
+
+const Input = styled.input`
+  padding: 5px;
+  border: 1px solid #ccc;
+  border-radius: 4px;
+`;
+
+const AnotherBadComponent: React.FC = () => {
+  const [userName, setUserName] = useState<string>('John Doe');
+  const [age, setAge] = useState<number>(25);
+  const [isPrimary, setIsPrimary] = useState<boolean>(true);
+
+  useEffect(() => {
+    console.log('Component did mount');
+  }, []);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUserName(e.target.value);
+  };
+
+  const togglePrimary = () => {
+    setIsPrimary(!isPrimary);
+  };
+
+  return (
+    <Container >
+      <div>
+        <Label>Username:</Label>
+        <Input value={userName} onChange={handleInputChange} />
+      </div>
+      <div>
+        <Label>Age:</Label>
+        <Input value={age.toString()} onChange={(e) => setAge(parseInt(e.target.value))} />
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/taulant-eulerity/Canva/pull/4#issuecomment-2181125499>#4bb06e</a><span> - 06/20/2024, 04:47 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'setAge' function directly parses the input value to an integer without any validation. This can lead to potential issues if the input is not a valid number.
</div>
<div id="fix">
<b>Fix:</b> Add validation to ensure the input value is a valid number before updating the state.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -52,7 +52,7 @@ const AnotherBadComponent: React.FC = () => {
    <Label>Age:</Label>
    <Input value={age.toString()} onChange={(e) => {
      const newAge = parseInt(e.target.value);
      if (!isNaN(newAge)) {
        setAge(newAge);
      }
    }} />
```
</div>
</details>
</div>
<br>


</div>